crates/protoc-gen-grafbase-subgraph/README.md

```diff
@@ -29,12 +29,63 @@ Make sure the binary is in your PATH, then run protoc with the `--grafbase-subgr
 protoc --grafbase-subgraph_out=. proto/*.proto
 ```
 
+## Custom options
+
+To make use of custom options, you will need to import the option definitions. They are defined in the `grafbase.options` package, available in the "options.proto" file in this project. Also note that since this module imports ["google/protobuf/descriptor.proto"](https://github.com/protocolbuffers/protobuf/blob/8228ee42b512cc330971e61bc9b86935a59f3477/src/google/protobuf/descriptor.proto), that one has to be present in your project as well.
+
+### Mapping RPC methods to Query fields
+
+```protobuf
+import "grafbase/options.proto";
+
+service SearchService {
+  rpc Search(SearchRequest) returns (SearchResponse) {
+    option (grafbase.options.is_graphql_query_field) = true;
+  }
+}
+```
+
+### Default all service methods to Query fields
+
+```protobuf
+import "grafbase/options.proto";
+
+service SearchService {
+  option (grafbase.options.graphql_default_to_query_fields) = true;
+
+  rpc Search(SearchRequest) returns (SearchResponse) {
+    option (grafbase.options.graphql_method_field_directives) = "@lookup";
+  }
+}
+```
+
+### Adding GraphQL directives on types, fields and enum values
+
+```protobuf
+import "grafbase/options.proto";
+
+message MyMessage {
+  option (grafbase.options.graphql_output_object_directives) = "@key(fields: \"id\")";
+
+  string id = 1 [(grafbase.options.graphql_output_field_directives) = "@deprecated"];
+}
+
+enum Color {
+  option (grafbase.options.graphql_enum_directives) = "@deprecated";
```

Comment:
I would have preferred grafbase.options.graphql_directives everywhere, but the options on each location are treated as separate items protoc, and they live in the same namespace, so that's not possible.

Reply:
Protoc fails, it says there's already an option named graphql_directives.

Reply:
ok, shame, but not a big deal